### PR TITLE
[INLONG-9064][Sort] Add Audit report for Pulsar connector in flink 1.15

### LINKED ISSUE #9064: [Feature] Add Audit report for Pulsar connector in flink 1.15
### Description

[Feature] Add Audit report for Pulsar connector in flink 1.15 

### Use case

[Feature] Add Audit report for Pulsar connector in flink 1.15 

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by EMsnap**
### Prepare a Pull Request
- Fixes #9064 

### Motivation

[INLONG-9064][Sort] Add Audit report for Pulsar connector in flink 1.15

### Modifications

[INLONG-9064][Sort] Add Audit report for Pulsar connector in flink 1.15

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

**Review by vernedeng — APPROVED** [review]
LGTM

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/pulsar/src/main/java/org/apache/inlong/sort/pulsar/PulsarTableOptions.java [hunk 1]
```diff
@@ -89,7 +89,7 @@ private PulsarTableOptions() {
                                     .build());
 
     public static final ConfigOption<String> SOURCE_START_FROM_MESSAGE_ID =
```
Comment (yunqingmoswu) [inline inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/pulsar/src/main/java/org/apache/inlong/sort/pulsar/PulsarTableOptions.java:91]: It is recommend to keep the meaning between the name of constant and the `ConfigOptions.key`

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/pulsar/src/main/java/org/apache/inlong/sort/pulsar/PulsarTableOptions.java [hunk 2]
```diff
@@ -282,4 +282,11 @@ private PulsarTableOptions() {
                     .booleanType()
                     .defaultValue(true)
                     .withDescription("Indicate if the table is an explicit Flink table.");
+
+    public static final ConfigOption<Boolean> GENERIC = ConfigOptions
```
Comment (yunqingmoswu) [inline inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/pulsar/src/main/java/org/apache/inlong/sort/pulsar/PulsarTableOptions.java:286]: What the meaning if false?Maybe it is needed a clearer definition.

**Review by e-mhui — APPROVED** [review]

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/pulsar/src/main/java/org/apache/inlong/sort/pulsar/PulsarTableOptions.java (hunk 2 above)
Comment (EMsnap) [reply inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/pulsar/src/main/java/org/apache/inlong/sort/pulsar/PulsarTableOptions.java]: there is no use of this parameter in flink 1.15 , so I removed it 
for flink1.13 its value is always true, so I removed it in flink 1.13 also to keep consistent

File: inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/pulsar/src/main/java/org/apache/inlong/sort/pulsar/PulsarTableOptions.java (hunk 1 above)
Comment (EMsnap) [reply inlong-sort/sort-flink/sort-flink-v1.15/sort-connectors/pulsar/src/main/java/org/apache/inlong/sort/pulsar/PulsarTableOptions.java]: the name changed already

**Review by yunqingmoswu — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**